File: 005.py
pfactors = [2, 3, 5, 7, 11, 13, 17, 19]
factor_number = [0]*8
for n in range(1, 20):
    for idx, val in enumerate(pfactors):
        j = 0
        m = n
        while m % val == 0:
            j += 1
            m = m/val
        if j > factor_number[idx]:
            factor_number[idx] = j

product = 1
for idx, val in enumerate(pfactors):
    product = product*val**factor_number[idx]
print product

File: 005/005.rs
fn main(){
    let pfactors: [u32; 8] = [2, 3, 5, 7, 11, 13, 17, 19];
    let mut factor_number: [u32; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
    for n in 1..=20{
        for (idx, val) in pfactors.iter().enumerate(){
            let mut j = 0;
            let mut m = n;
            while m % val == 0{
                j += 1;
                m = m / val;
            }
            if j > factor_number[idx]{
                factor_number[idx] = j;
            }
        }
    } 
    let mut product = 1;
    for (idx, val) in pfactors.iter().enumerate(){
        product *= (*val).pow(factor_number[idx]);
    }
    println!("{}", product);

}